subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -195,10 +189,8 @@ int bt_csip_sef(const uint8_t k[BT_CSIP_CRYPTO_KEY_SIZE], const uint8_t sirk[BT_
 
 	LOG_DBG("BE: k1 result %s", bt_hex(k1_out, sizeof(k1_out)));
 
-	if (IS_ENABLED(CONFIG_LITTLE_ENDIAN)) {
-		/* Swap result back to little endian */
-		sys_mem_swap(k1_out, sizeof(k1_out));
-	}
+	/* Get result back to host endianness. */
+	sys_be_to_cpu(k1_out, sizeof(k1_out));
```

Comment:
Same as above, the result of k1() is stored as MSO to LSO (big-endian byte order) but needs to be LSO to MSO (little-endian byte order) for the following xor to work.